ext/zend_test/test.c

```diff
@@ -531,7 +531,7 @@ PHP_MINIT_FUNCTION(zend_test)
 			1
 		);
 
-		ZVAL_STRINGL(&attr->args[0].value, "value1", sizeof("value1") - 1);
+		ZVAL_STRING(&attr->args[0].value, "value1");
```

Comment:
You want ZVAL_PSTRING() - persistent strings.

Reply:
I see, I'll just take some time later and fix it.

Reply:
Thank you! Feel free to take the tests I've written or to create a PR for the `WoltLab:function-attributes-leak` branch of our fork.

If you take the tests and write your own fix, then a `Co-authored-by: Tim Düsterhus <duesterhus@woltlab.com>` attribution would be appreciated.

Reply:
Actaully, that's just another issue with internal attributes:
https://github.com/php/php-src/commit/295ce7f19a266aa7b9072ae262a174453b47f5f7 (feel free to copy or cherry-pick)

Reply:
Thanks, I've taken that one. However now another assert is violated:

```
php: /home/timwolla/Schreibtisch/php-src/Zend/zend_variables.c:99: zval_internal_ptr_dtor: Assertion `(zval_gc_flags((str)->gc.u.type_info) & (1<<7))' failed.
==224624== 
==224624== Process terminating with default action of signal 6 (SIGABRT)
==224624==    at 0x4A4C15B: raise (raise.c:51)
==224624==    by 0x4A2B858: abort (abort.c:79)
==224624==    by 0x4A2B728: __assert_fail_base.cold (assert.c:92)
==224624==    by 0x4A3CF05: __assert_fail (assert.c:101)
==224624==    by 0x751644: zval_internal_ptr_dtor (zend_variables.c:99)
==224624==    by 0x78506E: attr_free (zend_attributes.c:218)
==224624==    by 0x7724E6: zend_hash_destroy (zend_hash.c:1667)
==224624==    by 0x73ED92: zend_hash_release (zend_hash.h:359)
==224624==    by 0x73F483: zend_function_dtor (zend_opcode.c:161)
==224624==    by 0x7725D6: zend_hash_destroy (zend_hash.c:1689)
==224624==    by 0x7542BA: compiler_globals_dtor (zend.c:740)
==224624==    by 0x692276: ts_free_id (TSRM.c:541)
==224624== 
```

Reply:
On that note (but out of scope of this PR) ... I'm wondering whether there's a time to relax the restrictions around valid values ... like internal code cannot use enum objects or arrays for neither of constants, property defaults, attribute parameters etc.. I'm totally aware that this is not so _simple_. But should be done...

Reply:
Figured it out: ZVAL_COPY_OR_DUP() -> ZVAL_DUP()